### PR TITLE
Push `dependencies` image to the new registry

### PR DESCRIPTION
**Pull request by kkovaletp**
This is the 1st of 2 PRs to migrate to the new Docker registry, where we have more control over the images and are able to manage them.
The new registry is https://hub.docker.com/u/photoview

### CODE REVIEW

File: .github/workflows/dependencies.yml [hunk 1]
```diff
@@ -10,11 +10,10 @@ on:
...
 
 env:
-  IS_PUSHING_IMAGES: ${{ github.event_name != 'pull_request' && github.repository == 'photoview/photoview' }}
+  #IS_PUSHING_IMAGES: ${{ github.event_name != 'pull_request' && github.repository == 'photoview/photoview' }}
+  IS_PUSHING_IMAGES: true
   IS_CACHING: ${{ github.event_name == 'pull_request' }}
-  DOCKER_USERNAME: viktorstrate
-  DOCKER_IMAGE: viktorstrate/dependencies
-  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  DOCKER_IMAGE: ghcr.io/photoview/dependencies
```
Comment (googollee) [inline .github/workflows/dependencies.yml:16]: I guess it should be `ghcr.io/photoview/photoview/dependencies`, as it's a image of `photoview/photoview` repo.

**Review by googollee — APPROVED** [review]
We can try this workflow first. If it fails, we could update the image name.

File: .github/workflows/dependencies.yml (hunk 1 above)
Comment (kkovaletp) [reply .github/workflows/dependencies.yml]: I've tried this change: https://github.com/photoview/photoview/actions/runs/11559527551/job/32174376849?pr=1108#step:7:23138
But it also doesn't work, as this is not a regular repo, but part of the `photoview` org. GitHub registry for images and packages is on the org level, not on the repo level for org repos, so the initial address was correct.
Anyway, I wanted to double-check it

Comment (googollee) [reply .github/workflows/dependencies.yml]: Did you grant the writing permission to workflows in Settings? 
image

Comment (kkovaletp) [reply .github/workflows/dependencies.yml]: I don't have admin permissions in this org(
@viktorstrate, this question to you

Comment (googollee) [reply .github/workflows/dependencies.yml]: I'd like to try the simple one first: `ghcr.io/photoview/photoview:dependencies` or even `ghcr.io/photoview/photoview`. If all settings are right, these should work well. After that, we can check how a sub-image works.

Comment (kkovaletp) [reply .github/workflows/dependencies.yml]:
```
Error: buildx failed with: ERROR: failed to solve: failed to push ghcr.io/photoview/photoview/photoview:latest: unexpected status from POST request to https://ghcr.io/v2/photoview/photoview/photoview/blobs/uploads/: 403 Forbidden
```
So, basically, we cannot use the GHCR of the Photoview org.

Comment (googollee) [reply .github/workflows/dependencies.yml]: I see you change the image name to `ghcr.io/photoview/photoview/photoview`. Is that right? I think it should be 'ghcr.io/photoview/photoview`.

Comment (kkovaletp) [reply .github/workflows/dependencies.yml]: I tried that earlier with the `dependencies` image. If GHCR accepts login and allows pushing, it should be a different error message

Comment (kkovaletp) [reply .github/workflows/dependencies.yml]: I've successfully pushed the image to the Docker hub org registry with my temporary PAT used. I've asked @viktorstrate in the Discord to double-check what is inside the current secret (PAT or password), as only the PAT is fully-supported.

**Review by viktorstrate — APPROVED** [review]
Let's try and see if this will work

**Comment by kkovaletp** [comment]
@googollee, @viktorstrate, after today's investigation I've found that the reason, why the secret doesn't work, is that this is an intentional implementation of GitHub Actions to prevent stealing secrets by submitting malicious PRs.
https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/
Long story short:
> Due to the dangers inherent to automatic processing of PRs, GitHub’s standard pull_request workflow trigger by default prevents write permissions and secrets access to the target repository.

So, there is only the way to review the PR and merge it with the hope that it will work in the `master` branch.

**Review by googollee — APPROVED** [review]

**Pull request merged**